README.md

```diff
@@ -31,167 +31,118 @@ Where `<snapshot version>` refers to the version in the Sonatype Snapshot badge
+  implicit val schema: Schema.CaseClass2[String, Int, Address] =
+    DeriveSchema.gen[Address]
 
-In order to get the functional effect of executing a specified Elasticsearch request, you must call the `execute` method defined on it, which returns a `ZIO` that requires an `ElasticExecutor`, fails with a `Throwable` and returns the relevant value `A` for that request.
-
-Elastic requests for creating and deleting return `CreationOutcome` and `DeletionOutcome` respectively if no other meaningful value could be returned, notifying us on the success of the request. Any other kind of error is returned as a `Throwable` in the error channel of `ZIO` for that Elastic request.
-
-If you want to execute multiple Elasticsearch requests in a single API call, you need to use the `bulk` method on those Elastic requests, and call `execute` on that bulk request instead.
-
-To provide the dependency on `ElasticExecutor`, you must pass one of the `ZLayer`s from the following import:
-
-```scala
-import zio.elasticsearch.ElasticExecutor
-```
+  val (street, number) = schema.makeAccessors(FieldAccessorBuilder)
+}
 
-For example, if you want to execute requests on a server running on `localhost` and port `9200`, you can provide the `live` ZLayer to your effect, along with a `SttpBackend` and an `ElasticConfig` layer:
+final case class User(id: Int, address: Address)
 
-```scala
-import sttp.client3.SttpBackend
-import sttp.client3.httpclient.zio.HttpClientZioBackend
-import zio.elasticsearch.{ElasticConfig, ElasticExecutor}
+object User {
+  implicit val schema: Schema.CaseClass2[String, Address, User] =
+    DeriveSchema.gen[User]
 
-val effect: RIO[ElasticExecutor, Boolean] = exists(IndexName("index"), DocumentId("document")).execute
+  val (id, address) = schema.makeAccessors(FieldAccessorBuilder)
+}
 
-effect.provide(
-  HttpClientZioBackend.layer(),
-  ZLayer.succeed(ElastichConfig("localhost", 9200)) >>> ElasticExecutor.live,
-)
+for {
+  _ <- Elasticsearch.execute(
+    search(
+      IndexName("index"),
+      must(range(User.id).gte(7).lt(10)).should(startsWith(User.address / Address.street, "ZIO"))
+    ).aggregate(aggregation)
+  )
+} yield ()
 ```
 
-If the ElasticConfig arguments are the same as specified above, you can simply omit the `ElasticConfig` layer and replace `ElasticExecutor.live` with `ElasticExecutor.local` instead.
+### Streaming
 
-For testing purposes, you can use `ElasticExecutor.test`, which is a mocked Elasticsearch executor that doesn't require an HTTP backend.
+ZIO Elasticsearch is a streaming-friendly library, and it provides specific APIs for creating ZIO streams. When using the stream API, the result will be an `Item`, which is a case class that contains only one field, `raw`, that represents your response as raw JSON. Additionally, it is important to note that you can use `StreamConfig` to customize your settings when creating a stream. If you don't use `StreamConfig`, the default settings (`StreamConfig.Default`) will be used.
```

Comment:
In this paragraph we're explaining `Item` and `raw`, while I believe the more common use case would be the typed `streamAs[A]`, which we're not mentioning, so perhaps we should change it to mention both, with `streamAs[A]` being in focus and `raw` being an additional option.